src/daft-local-execution/src/pipeline.rs

```diff
@@ -282,17 +283,63 @@ pub fn physical_plan_to_pipeline(
             group_by,
             pivot_column,
             value_column,
+            aggregation,
             names,
             ..
         }) => {
-            let pivot_op = PivotOperator::new(
+            let child_node = physical_plan_to_pipeline(input, psets, cfg)?;
+            // Pivot is a two stage operation. The first stage aggregates the values, grouped by both the group by columns and the pivot column.
+            // The second stage is a pivot operation where the values are pivoted into columns based on the pivot column.
+
+            let group_by_with_pivot = group_by
+                .iter()
+                .chain(std::iter::once(pivot_column))
+                .cloned()
+                .collect::<Vec<_>>();
+            let aggregate_schema = || -> DaftResult<_> {
+                let fields = group_by_with_pivot
+                    .iter()
+                    .map(|expr| expr.to_field(input.schema()))
+                    .chain(std::iter::once(aggregation.to_field(input.schema())))
+                    .collect::<DaftResult<Vec<_>>>()?;
+                Schema::new(fields)
+            }()
```

Comment:
Why is this wrapped as a function and then immediately called?

Reply:
So I can add the `with_context` only once